sdk/clientcore/http-okhttp3/src/main/java/io/clientcore/http/okhttp3/implementation/ProxyAuthenticator.java

```diff
@@ -84,76 +95,90 @@ public Interceptor getProxyAuthenticationInfoInterceptor() {
      */
     @Override
     public Request authenticate(Route route, Response response) {
-        String authorizationHeader =
-            challengeHandler.attemptToPipelineAuthorization(PROXY_METHOD, PROXY_URI_PATH, NO_BODY);
-
-        // Pipelining was successful, use the generated authorization header.
-        if (!isNullOrEmpty(authorizationHeader)) {
-            return response.request().newBuilder()
-                .header(PROXY_AUTHORIZATION, authorizationHeader)
-                .build();
-        }
-
-        // If this is a pre-emptive challenge quit now if pipelining doesn't produce anything.
+        // Handle preemptive authentication
         if (PREEMPTIVE_AUTHENTICATE.equalsIgnoreCase(response.message())) {
+            // If we have already authenticated, apply the stored Proxy-Authorization header.
+            if (lastAuthorizationHeader != null) {
+                return response.request().newBuilder()
+                    .header(PROXY_AUTHORIZATION, lastAuthorizationHeader)
+                    .build();
+            }
+            // If no previous authorization, return the request unchanged.
             return response.request();
         }
 
-        boolean hasBasicChallenge = false;
-        List<Map<String, String>> digestChallenges = new ArrayList<>();
-
-        for (Challenge challenge : response.challenges()) {
-            if (BASIC.equalsIgnoreCase(challenge.scheme())) {
-                hasBasicChallenge = true;
-            } else if (DIGEST.equalsIgnoreCase(challenge.scheme())) {
-                digestChallenges.add(challenge.authParams());
+        Request.Builder requestBuilder = response.request().newBuilder();
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.valueOf(PROXY_METHOD), PROXY_URI_PATH);
+        HttpResponse<?> httpResponse = new HttpResponse<>(httpRequest, response.code(), OkHttpResponse.fromOkHttpHeaders(response.headers()), NO_BODY);
+        String authorizationHeader;
+        // Replace nonce value in the PROXY_AUTHENTICATE header with the updated nonce
+        ConcurrentHashMap<String, String> lastChallengeMap = proxyInterceptor.getLastChallenge();
+        String updatedNonce = lastChallengeMap.get(NONCE);
+
+        if (updatedNonce != null) {
+            String proxyAuthenticateHeader = httpResponse.getHeaders().get(HttpHeaderName.PROXY_AUTHENTICATE).getValue();
+
+            if (proxyAuthenticateHeader != null) {
+                // Replace the old nonce with the updated nonce in the header
+                String updatedHeader = replaceNonceInHeader(proxyAuthenticateHeader, updatedNonce);
```

Comment:
Why are we replacing the nonce in the Proxy-Authenticate header?

Reply:
I think we want to keep this capability to avoid "replay attacks" as explained here [RFC 2069](https://pages.cs.wisc.edu/~cao/digest-rfc2069.html). The server issues a new nonce periodically to prevent old authentication responses from being reused. Replacing the nonce with the updated value helps meet this security requirement.